### DISCUSSION TITLE
IGNORE / INCORRECT: FYI - rewrite-maven-plugin won't run under Java 8 as of today's release of v5.34.0

### DISCUSSION THREAD

**Discussion by gjd6640** [openrewrite/rewrite#4272] [General] [answered]
Just a heads-up:  Anyone whose JAVA_HOME is set to a Java 8 installation will fail with the below error when they use today's release of rewrite-maven-plugin. You'll need to adjust JAVA_HOME to a modern JVM version to use it.

```
[ERROR] Failed to execute goal org.openrewrite.maven:rewrite-maven-plugin:5.34.0:run (default-cli) on project ntp-timed-
job-drifted-trains-identifier: Execution default-cli of goal org.openrewrite.maven:rewrite-maven-plugin:5.34.0:run faile
d: Could not load class ai.timefold.solver.migration.fork.TimefoldChangeDependencies : java.lang.UnsupportedClassVersion
Error: ai/timefold/solver/migration/fork/TimefoldChangeDependencies has been compiled by a more recent version of the Ja
va Runtime (class file version 61.0), this version of the Java Runtime only recognizes class file versions up to 52.0 ->
 [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.openrewrite.maven:rewrite-maven-plugi
n:5.34.0:run (default-cli) on project ntp-timed-job-drifted-trains-identifier: Execution default-cli of goal org.openrew
rite.maven:rewrite-maven-plugin:5.34.0:run failed: Could not load class ai.timefold.solver.migration.fork.TimefoldChange
Dependencies : java.lang.UnsupportedClassVersionError: ai/timefold/solver/migration/fork/TimefoldChangeDependencies has
been compiled by a more recent version of the Java Runtime (class file version 61.0), this version of the Java Runtime o
nly recognizes class file versions up to 52.0
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:213)
```

This may have been an intentional change. Just wanted to make sure that the community is aware of the change.

**Comment by gjd6640** [comment]
I spoke too soon. Looks like a recipe that I added is what's leading to this error.

**Reply by timtebeek** [reply]
Indeed looks like `TimefoldChangeDependencies` might be the one that requires a newer JDK. That comes in through https://github.com/openrewrite/rewrite-third-party ; we might need to revise that, as it was added to rewrite-testing-libraries with this release: https://github.com/openrewrite/rewrite-testing-frameworks/commit/60cff4b0bfdd8c426be9815d552f938b6644aea9

**Reply by timtebeek** [reply]
Could you clarify if you added rewrite-third-party explicitly, or if that came in transitively with rewrite-testing-frameworks?

**Reply by timtebeek** [reply, marked as answer]
You're right, and thanks for pointing this out! I've logged an issue and will revert the change in rewrite-testing-frameworks that pulled in the Java 17 classes, and then do the downstream releases. :(
- https://github.com/openrewrite/rewrite-third-party/issues/11
- https://github.com/openrewrite/rewrite-testing-frameworks/pull/531
- https://github.com/openrewrite/rewrite-testing-frameworks/releases/tag/v2.12.1

The fix has landed in
- https://github.com/openrewrite/rewrite-recipe-bom/releases/tag/v2.13.1
